src/dlite-json.c

```diff
@@ -1076,7 +1076,7 @@ DLiteJsonFormat dlite_jstore_loads(JStore *js, const char *src, int len)
   unsigned int ntokens=0;
   char uuid[DLITE_UUID_LENGTH+1], *uri=NULL;
   int r;
-  DLiteJsonFormat format=-1;
+  DLiteJsonFormat retval=-1, format;
```

Comment:
Not very clear to me what this new 'format' variable is, why it has been added or what it does.

Reply:
The function returns the json `format` or -1 on error.  Before there was a possibility that this function returned a json format, despite there was an error. This bug was corrected by introducing the `retval` variable which is initialised to -1 and only set to a (non-negative) format value if the function succeeds.